cartridges/int_klaviyo_core/cartridge/templates/default/klaviyo/klaviyoListeners.isml

```diff
@@ -107,3 +109,40 @@
         klaviyoListeners();
     </script>
 </isif>
+
+<isif condition="${KLEmailListID || KLSmsListID}">
+    <script>
+        klaviyoJS.subscribe = function(data) {
+            $.ajax({
+                url: "${dw.web.URLUtils.https('Klaviyo-Subscribe')}",
+                type: 'post',
+                data: data,
+                error: function (err) {
+                    console.warn('Klaviyo Subscribe failure: ' + err.statusText);
+                }
+            });
+        }
+
+        window.addEventListener('DOMContentLoaded', (event) => {
+            klaviyoJS.emailSubscribe = document.getElementById('KLEmailSubscribe');
```

Comment:
If a user fails to add the snippets to the template which will render the KLEmailSubscribe checkbox, will this be an issue?

Reply:
I don't think it is from looking here but wanted to sanity check myself

Reply:
Great question! No, an error would not be thrown if a user forgot to add the snippets. Issues are avoided because of conditionals that check for those values before any other logic is run. For example, `klaviyoJS.emailSubscribe` stores the result of `document.getElementById('KLEmailSubscribe');`

If a user fails to add the snippets, then that query will result in a null value. So when we check to see whether the `klaviyoJS.emailSubscribe` has a value, the null check would prevent other logic from being run on that property. It's the same handling for the email & sms properties that are stored on the klaviyoJS object within this template.